res/schema.xml

```diff
@@ -489,4 +489,29 @@ METADATA
       ALTER TABLE library ADD COLUMN color INTEGER;
     </sql>
   </revision>
+  <revision version="32" min_compatible="3">
+    <description>
+      Convert the PredefinedColor ID to the actual RGBA value.
+    </description>
+    <sql>
+      <!-- No Color becomes white-->
+      UPDATE cues SET color=0xFFF2F2FF WHERE color=0;
```

Comment:
I think we need to decide on the default color now before merging this database schema update. Also, this should match the default color, which it doesn't now. Currently the default color is red.

I agree that using red, green, or blue as the default color isn't a great choice. They don't necessarily stand out from the RGB waveforms. I am not sure about white either though. It doesn't always stand out very much. I think cyan (#42D4F4) stand out from the waveforms and from other skin colors without drawing too much attention.

Reply:
@Holzhaus Although we agreed on the _no color_ use case: Would it help to make the color in the database (and maybe even in the `Cue` class) optional and do the mapping to a default color on a higher level, i.e. like you have already successfully implemented for track colors? If this is the case I could provide a schema migration to make the column _nullable_, otherwise just ignore this comment ;)

Reply:
@uklotzde
> Would it help to make the color in the database (and maybe even in the `Cue` class) optional and do the mapping to a default color on a higher level, i.e. like you have already successfully implemented for track colors?

No, but thanks for the offer ;-) Track colors are different since they are only an addition highlight (i.e. tracks may have no color), but we already agreed that *all* hotcues have a color (since we need to color the hotcue buttons anyway to distinguish set hotcues from unset hotcues). And I'd rather handle this at  the DB level.

Reply:
@Be-ing 

> I think we need to decide on the default color now before merging this database schema update. Also, this should match the default color, which it doesn't now. Currently the default color is red.

Why? Wouldn't it be nice to be able to distinguish legacy cues from new cues easily? I used white because it should look good in all skins. Cyan would probably look good in Deere, but ugly in LateNight.

Anyway, while I'd prefer "white", I'm actually pretty dispassionate about this, because there'll be a mass-color change tool anyway. Just make a decision either here or on Zulip and tell me what I should do ;-)

> I agree that using red, green, or blue as the default color isn't a great choice. They don't necessarily stand out from the RGB waveforms. I am not sure about white either though. It doesn't always stand out very much. I think cyan (#42D4F4) stand out from the waveforms and from other skin colors without drawing too much attention.

If we want to make hotcues easily distinguishable, we could also  "auto_hotcue_colors" by default. Users that don't like this can still turn it off before starting to set new cues after upgrade. Coloring hotcues in different colors based on the hotcue number is also what Serato does by default. Users that care about their hotcue colors will set them manually anyway, and users that don't care get easily distinguishable hotcues for free (i.e. no opt-in needed). But again, I'm okay with both options.